turbinia/evidence.py

```diff
@@ -38,6 +41,65 @@
 log = logging.getLogger('turbinia')
 
 
+def evidence_class_names(all_classes=False):
+  """Returns a list of class names for the evidence module.
+
+  Args:
+    all_classes (bool): Flag to determine whether to include all classes
+        in the module.
+  """
+  predicate = lambda member: inspect.isclass(member) and not inspect.isbuiltin(
+      member)
+  class_names = inspect.getmembers(sys.modules[__name__], predicate)
+  if not all_classes:
+    # TODO: Non-evidence types should be moved out of the evidence module,
+    # so that we no longer have to ignore certain classes here. Especially
+    # 'output' and 'report' types.
+    # Ignore classes that are not real Evidence types and the base class.
+    ignored_classes = (
+        'BinaryExtraction', 'BodyFile', 'BulkExtractorOutput', 'Evidence',
+        'EvidenceState', 'EvidenceCollection', 'ExportedFileArtifact',
+        'FilteredTextFile', 'FinalReport', 'IntEnum', 'PlasoCsvFile',
+        'PlasoFile', 'PhotorecOutput', 'ReportText', 'TextFile',
+        'VolatilityReport', 'TurbiniaException')
+    class_names = filter(
+        lambda class_tuple: class_tuple[0] not in ignored_classes, class_names)
+  return list(class_names)
+
+
+def map_evidence_attributes():
+  """Creates a dictionary that maps evidence types to their
+       constructor attributes.
+
+  Returns:
+    object_attribute_mapping (defaultdict): A mapping of evidence types
+        and their constructor attributes.
+  """
+  object_attribute_mapping = defaultdict(list)
+  for class_name, class_type in evidence_class_names():
+    try:
+      attributes_signature = inspect.signature(class_type)
+      attributes = attributes_signature.parameters.keys()
+      for attribute in attributes:
+        if not object_attribute_mapping[class_name]:
+          object_attribute_mapping[class_name] = defaultdict(dict)
+        # Ignore 'args' and 'kwargs' attributes.
+        if attribute not in ('args', 'kwargs'):
+          object_attribute_mapping[class_name][attribute] = {
+              'required': bool(attribute in class_type.REQUIRED_ATTRIBUTES),
+              'type': 'str'
```

Comment:
Any other thoughts on how to handle typing for these?  I think currently some of these things get passed in as other types (e.g. `int`).  Should we just assume that everything passed in is a string and leave it up to the Evidence objects to verify and cast them correctly?  Or is there something more we can do with type hinting?  If we are assuming the evidence objects will handle them, we should probably document that somewhere and also go through them (if we haven't already) and make sure all of the parameters are handled correctly.

Reply:
Yeah, type annotations will help with this. Without that, I couldn't find a way to infer the types. I'm happy to go back to this and make adjustments once we introduce type hinting to the module. I think we should probably do that in a separate PR though. I will make a note of this in #1095

Reply:
Done, I made a note in the type hinting issue about this. In the meantime, try/except blocks to cast to int.